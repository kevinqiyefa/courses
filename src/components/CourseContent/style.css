.course-btns {
  text-align: left;
  margin-top: 3rem;
}

.course-btns > button {
  width: 9rem;
  margin-right: 0.5rem;
  font-weight: bold;
}
