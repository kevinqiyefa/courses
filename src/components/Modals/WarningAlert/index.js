import React from 'react';
import PropTypes from 'prop-types';

function WarningAlert({ message, deleteCourse }) {
  return (
    <div
      className="modal fade"
      id="deleteWarning"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="deleteWarningLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog modal-dialog-centered" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              className="modal-title"
              id="deleteWarningLabel"
              style={{ color: 'red' }}
            >
              Warning
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">{message}</div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              className="btn btn-danger"
              data-dismiss="modal"
              onClick={deleteCourse}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

WarningAlert.propTypes = {
  message: PropTypes.string,
  deleteCourse: PropTypes.func
};

export default WarningAlert;
