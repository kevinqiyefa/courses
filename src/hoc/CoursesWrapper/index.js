import React from 'react';
import PropTypes from 'prop-types';

function CoursesWrapper({ title, description, children }) {
  return (
    <div className="jumbotron container mt-5">
      <h1 className="display-3">{title}</h1>
      <p className="lead mb-5">{description}</p>
      {children}
    </div>
  );
}

CoursesWrapper.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default CoursesWrapper;
