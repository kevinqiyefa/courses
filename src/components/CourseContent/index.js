import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import CoursesContainer from '../../hoc/CoursesWrapper';
import './style.css';
import WarningAlert from '../WarningAlert';
import AddEditCourse from '../AddEditCourse';

class CourseContent extends Component {
  state = {
    subject: 'English',
    length: 45,
    course_name: '',
    course_description: '',
    studentID: [],
    isInitialed: false
  };

  static getDerivedStateFromProps(props, state) {
    const {
      subject,
      length,
      course_name,
      course_description,
      studentID
    } = props.course;
    if (!state.isInitialed) {
      return {
        subject,
        length,
        course_name,
        course_description,
        studentID,
        isInitialed: true
      };
    }

    // Return null if the state hasn't changed
    return null;
  }

  handleInputChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleSliderChange = value => {
    this.setState({ length: value });
  };

  handleSubmit = () => {
    const { isInitialed, ...updatedCourse } = this.state;

    this.props.patchCourse(updatedCourse);
  };

  render() {
    const { course, deleteCourse } = this.props;
    const warningMsg = 'Are you sure you want to do this?';

    const modalTitle = 'Editing a course!';
    const submitBtnText = 'Edit Course';

    return (
      <CoursesContainer
        title={course.course_name}
        description={course.course_description}
      >
        <div className="course-card-subject">
          <h5>Subject: {course.subject}</h5>
          <h5>Length: {course.length} mins</h5>
        </div>
        <div className="course-btns">
          <button type="button" className="btn btn-success">
            <Link to="/224/students">View Students</Link>
          </button>
          <button
            type="button"
            className="btn btn-warning"
            data-toggle="modal"
            data-target="#addEditCourseModal"
          >
            Edit
          </button>
          <button
            type="button"
            className="btn btn-danger"
            data-toggle="modal"
            data-target="#deleteWarning"
          >
            Delete
          </button>
        </div>
        <WarningAlert message={warningMsg} deleteCourse={deleteCourse} />
        <AddEditCourse
          state={this.state}
          handleInputChange={this.handleInputChange}
          handleSliderChange={this.handleSliderChange}
          handleSubmit={this.handleSubmit}
          modalTitle={modalTitle}
          submitBtnText={submitBtnText}
        />
      </CoursesContainer>
    );
  }
}

CourseContent.propTypes = {
  course: PropTypes.object,
  deleteCourse: PropTypes.func,
  patchCourse: PropTypes.func
};

export default CourseContent;
