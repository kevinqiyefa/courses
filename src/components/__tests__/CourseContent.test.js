import React from 'react';
import { shallow } from 'enzyme';
import CourseContent from '../CourseContent';
import CoursesWrapper from '../CoursesWrapper';
import WarningAlert from '../Modals/WarningAlert';

import toJson from 'enzyme-to-json';

describe('Testing CourseContent Component', () => {
  let wrapped;
  const coursesData = {
    id: 111,
    subject: 'Math',
    length: 60,
    course_name: 'Algebra I',
    course_description:
      'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
    studentIDs: [2, 5]
  };

  beforeEach(() => {
    wrapped = shallow(<CourseContent course={coursesData} />);
  });

  it('shows a CoursesWrapper section', () => {
    expect(wrapped.find(CoursesWrapper).length).toEqual(1);
  });

  it('shows a WarningAlert modal', () => {
    expect(wrapped.find(WarningAlert).length).toEqual(1);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapped);
    expect(serialized).toMatchSnapshot();
  });
});
