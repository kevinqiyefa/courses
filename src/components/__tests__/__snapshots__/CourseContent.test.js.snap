// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CourseContent Component matches snapshot 1`] = `
<CoursesWrapper
  description="Lorem Ipsum is simply dummy text of the printing and typesetting industry."
  title="Algebra I"
>
  <div
    className="course-card-subject"
  >
    <h5>
      Subject: 
      Math
    </h5>
    <h5>
      Length: 
      60
       mins
    </h5>
  </div>
  <div
    className="course-btns"
  >
    <Link
      to="/111/students"
    >
      <button
        className="btn btn-success"
        type="button"
      >
        View Students
      </button>
    </Link>
    <button
      className="btn btn-warning"
      data-target="#addEditCourseModal"
      data-toggle="modal"
      type="button"
    >
      Edit
    </button>
    <button
      className="btn btn-danger"
      data-target="#deleteWarning"
      data-toggle="modal"
      type="button"
    >
      Delete
    </button>
  </div>
  <WarningAlert
    message="Are you sure you want to do this?"
  />
  <AddEditCourse
    handleInputChange={[Function]}
    handleSliderChange={[Function]}
    handleSubmit={[Function]}
    modalTitle="Editing a course!"
    state={
      Object {
        "course_description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry.",
        "course_name": "Algebra I",
        "id": 111,
        "length": 60,
        "studentIDs": Array [
          2,
          5,
        ],
        "subject": "Math",
      }
    }
    submitBtnText="Edit Course"
  />
</CoursesWrapper>
`;
